### ISSUE TITLE
Add `react-aria` for accessibility to components.

### ISSUE THREAD

**Issue by GuptaSiddhant** [DPC-Nordics/dpc-next#13]
React ARIA library provided useful hooks to add accessibility props to components.
https://react-spectrum.adobe.com/react-aria

For example

- [`useButton`](https://react-spectrum.adobe.com/react-aria/useButton.html) hook will add required keyboard interaction and aria-* props for the button component. 
- Keyboard interaction includes using "space" and "return/enter" to trigger the click-event (which is called `onPress` for standardisation)
- Also, adding these hook will add accessibility warnings. `useButton` warns if the button is missing a label (in an icon-only button) and suggests that an `aria-label` should be provided.
